packages/ethash/package.json

```diff
@@ -38,6 +38,7 @@
   },
   "devDependencies": {
     "@ethereumjs/block": "^3.4.0",
+    "@ethereumjs/common": "^2.4.0",
```

Comment:
hmm do common and block need to be regular dependencies since they're used in the source now? this might be a good use for peerDependencies.

Reply:
Oh right, it is also in the source, so it should not be a devDependency. What exactly is a peer dependency? Have never encountered this.

Reply:
`Common` is not in the source - unless I am overlooking something - there is just one place where it is taken from a `block` instance.

`Block` is in the source, but it actually seems that `Block` types had been in the source before? 🤔 So this would rather be a bug fix I guess?

Not sure, I would tend to not start experimenting with peer dependencies now here on the side line but rather do in the "classic" way it is currently done in all other libraries and then execute on a common strategy (if we decide to) somewhat later.

Reply:
@holgerd77 ah right thanks for thinking it through. sounds good.

I believe `Block` was fine as a devDep when it was just used for its types because it would have only been needed at build time for creating the `.d.ts` files.

So here I think we can move `@ethereumjs/block` to a normal dep, and common can remain a devDep.

> What exactly is a peer dependency? Have never encountered this.

@jochem-brouwer check out #1200 for a deep dive on the issue. [This article](https://lexi-lambda.github.io/blog/2016/08/24/understanding-the-npm-dependency-model/) sums it up well, which is basically: if the dep is used in the exported public API then when specified as a peerDep it will be more compatible with varying versions in node_modules from other installed libraries since it will opt to use those first before installing its own, so it reduces finicky type mismatches from very small differences in types.

Reply:
Ok, I updated this according to the discussion. Also re-checked our [dependency relationship diagram](https://github.com/ethereumjs/ethereumjs-monorepo#package-dependency-relationship) to make sure that we are not introducing a circular dependency here.